File: scripts/update_sitemap.js
const fs = require('fs');

const sitemapPath = 'public/sitemap.xml';
let content = fs.readFileSync(sitemapPath, 'utf8');

const newEntry = `  <url>
    <loc>https://www.amalegalsolutions.com/default-loan-settlement-in-india</loc>
    <lastmod>2026-06-26</lastmod>
    <priority>0.8</priority>
  </url>
`;

if (!content.includes('default-loan-settlement-in-india')) {
  content = content.replace('</urlset>', newEntry + '</urlset>');
  fs.writeFileSync(sitemapPath, content);
  console.log('Sitemap updated.');
} else {
  console.log('Already in sitemap.');
}

File: scratch/update_sitemap.py
import re

files_to_update = [
    "/Users/amalegalsolutions/Desktop/AMAWORK/AMA/ama/public/sitemap.xml",
    "/Users/amalegalsolutions/Desktop/AMAWORK/AMA/ama/src/app/sitemap.xml"
]

entry = """
    <url>
      <loc>https://www.amalegalsolutions.com/loan-settlement-request-letter</loc>
      <lastmod>2026-06-30T12:00:00.000Z</lastmod>
      <changefreq>weekly</changefreq>
      <priority>0.8</priority>
    </url>
"""

for filepath in files_to_update:
    try:
        with open(filepath, "r") as f:
            content = f.read()
        
        if "loan-settlement-request-letter" not in content:
            content = content.replace("</urlset>", f"{entry}</urlset>")
            with open(filepath, "w") as f:
                f.write(content)
            print(f"Updated {filepath}")
        else:
            print(f"Already in {filepath}")
    except FileNotFoundError:
        print(f"File not found: {filepath}")

